ONSdigital/rm-case-service
# Add delete by ce endpoint

# What and why?
With the broken BRES collection exercise, we needed a way to delete the unusable cases in the database.
EDIT:
Modified PR to now work with a cascade on FKs from  casegroup allowing the ability to directly delete from collection exercise id hopefully allowing the sql server to carry the load and not overload case service making it unusual if a large collection was deleted during business hours. 
Corrected url so collection exercise isn't plural

# How to test?
Deploy this to dev, run acceptance tests to get quick case to delete or create one yourself. 
Go to database and find your case, Case Group table easy place to find collection exercise id, you will want to open Case and Case Event since you will be checking these later.
Using postman DELETE with http://localhost:8171/casegroups/collectionExercise/{collection_excercise_id}
Check database tables (Case Group, Case and Case Event) to see that case was deleted and nothing was left associated with that case and log to ensure no errors (there shouldn't be errors).
Check outputs and logs for a collection that exists and doesn't exist. 

# Jira
https://jira.ons.gov.uk/browse/RAS-283
